src/main/java/fi/aalto/cs/apluscourses/dal/APlusExerciseDataSource.java

```diff
@@ -195,6 +196,15 @@ public Exercise getExercise(long exerciseId,
     return parser.parseExercise(response, points, optionalCategories, tutorials);
   }
 
+  @Override
+  @NotNull
+  public String getSubmissionFeedback(long submissionId,
+                                      @NotNull Authentication authentication) throws IOException {
+    var url = apiUrl + "submissions/" + submissionId + "/";
```

Comment:
Nit: We have a string constant for `"submissions"`. Using those keeps it easier to track where certain API endpoints are called.